### DISCUSSION TITLE
What's the difference between Chatflow and Agentflow in Flowise?

### DISCUSSION THREAD

**Discussion by rui-vas** [FlowiseAI/Flowise#3013] [Q&A]
I'm new to Flowise and have been playing around with the tool while writing an article about my experience.

I see that I can create a `Chatflow` or a `Agentflow`:
![image](https://github.com/user-attachments/assets/30588c3e-4328-492b-85cc-d04e8c1555e0)

I also observe that the options in the menus are different. 

In ChatFlow:
![image](https://github.com/user-attachments/assets/603cbc2a-dedd-4305-ba39-6d801205b0fb)

In AgentFlow:
![image](https://github.com/user-attachments/assets/2fb39b54-5ad1-4602-8dbd-541b479269c7)

What is the rationale behind the difference, and when should I pick one versus the other? 

Many thanks!! :)

**Comment by taowang1993** [comment]
Consider "chatflow" as one sport player.
Consider "agentflow" as a team of players working together.

**Reply by ThangLuuQuoc** [reply]
And in real test I see a difference, when I use $flow.state.xxx example $flow.state.todoList to save my state in customtool. 
In chatfflow $flow.state it alway renew so $flow.state.todoList alway reset every new chat. and in agentflow my todolist realy saved and continue add new item in new chat message. And I just see State Node in agentflows, not see it in chatflow
